transcom/mymove
# B 21011 21012 int

## [B-21011](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21011) and [B-21012](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21012)

## Summary
These two agility backlog items add an additional page to the SSW for AOA and Payment Packets. It's an additional shipments page to keep track of all other shipments attached to the move. It has the same look as the other pages with rows for shipment information for up to 16 additional shipments (if more are supplied, the packet will not be created).

The values that display for the shipments can depend on shipment type, so please be sure to check them thoroughly when testing.

Additionally, please feel free to try to break the PDF generation process in other ways. This task added many potential redundant values that will most often be empty, so I created a new map and added on to the existing reflection system to make things look nicer and run smoother. I'm confident in it's stability, but just in case.

> [!WARNING]  
> Commits 2110b0f507fc61abab9070342a61b71945b62036 and 0c1eb5e2f02582d40461842b4dbf3abcd7b584be were merge commits required to resolve merge conflicts. If additional conflicts arise with a main PR for 21011, 21012, or [20947](https://github.com/transcom/mymove/pull/13594) , please reference these commits for resolutions.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a service member. Create a move with many shipments including at least 1 ppm with an advance. We need to make a PPM with advance to generate the AOA, and a completed PPM for the Payment Packet.
2. Submit the move with many shipments.
3. Log in as a service counselor, approve the advance request in order to unlock the AOA packet. Add more information to other shipments as you see fit for testing.
4. Download the AOA. Ensure the packet downloads, the third page is present and looks nice with preparation date and shipment information NOT including the PPM you generated the packet for (that information is shown on the first page)
5. Ensure the PPM move details have been submitted.
6. Logged back in as a service member, upload your documents for your PPM.
7. As a service counselor, approve the documents, and generate the payment packet.
8. Ensure all data is present in it's most updated form. 
9. Ensure the watermark on orders and other documents isn't present on SSW pages.
10. Ensure the tab name in your browser is "Shipment Summary Worksheet" (used to be SSWPDFTemplate)
11. Ensure all tests passing.

### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="549" alt="image" src="https://github.com/user-attachments/assets/a4b52343-a215-4e5f-a379-2c89481c8c2d">
